ckanext/privatedatasets/plugin.py

```diff
@@ -341,18 +341,22 @@ def before_view(self, pkg_dict):
 
         for resource in pkg_dict['resources']:
 
-            context = {
-                'model': model,
-                'session': model.Session,
-                'user': tk.c.user,
-                'user_obj': tk.c.userobj
-            }
-
             try:
-                tk.check_access('resource_show', context, resource)
-            except tk.NotAuthorized:
-                pkg_dict['resources'].remove(resource)
-                pkg_dict = self.before_view(pkg_dict)
+                context = {
+                    'model': model,
+                    'session': model.Session,
+                    'user': tk.c.user,
+                    'user_obj': tk.c.userobj
+                }
+
+                try:
+                    tk.check_access('resource_show', context, resource)
+                except tk.NotAuthorized:
+                    pkg_dict['resources'].remove(resource)
+                    pkg_dict = self.before_view(pkg_dict)
+            except:
+               print ("problem fixed")
```

Comment:
Do we need this print here?

Reply:
We need something to fill the except and state the problem has occurred (Will help with debugging future occurrence of this problem). I will change to a better print message